packages/cysync-core/src/dialogs/WalletConnect/Dialogs/AccountSelection.tsx

```diff
@@ -0,0 +1,165 @@
+    selectedEvmAccountsGroup,
+    selectedEvmAccounts,
+    handleWalletChange,
+    setSelectedEvmAccounts,
+    evmAccountsGroup,
+    dapp,
+    supportedNoAccountBlockchain,
+  } = useWalletConnectDialog();
+
+  return (
+    <DialogBox width={500} $maxHeight="90vh" align="stretch">
+      <Flex direction="row" justify="flex-end" py={2} px={3}>
+        <CloseButton onClick={onClose} />
+      </Flex>
+      <Divider variant="horizontal" />
+      <Container display="flex" direction="column" gap={32} py={4} px={5}>
+        <DappConnectedLogo
+          logos={[dapp.logo, WalletConnectLogo, cysyncLogoSmall]}
+        />
+        <Container display="flex" direction="column" gap={8} width="full">
+          <Typography variant="h5" $textAlign="center">
+            <LangDisplay
+              text={accountSelectionTab.title}
+              variables={{ dappName: dapp.name }}
+            />
+          </Typography>
+          <Typography variant="span" color="muted">
+            <LangDisplay text={dapp.url} />
+          </Typography>
+        </Container>
+      </Container>
+      <ScrollableContainer>
+        <DialogBoxBody pt={2} px={5} pb={4} align="stretch" gap={24}>
+          <Dropdown
+            items={walletDropdownList}
+            selectedItem={selectedWallet?.__id}
+            searchText={accountSelectionTab.chooseWallet}
+            placeholderText={accountSelectionTab.chooseWallet}
+            onChange={handleWalletChange}
+            leftImage={
+              <Image
+                src={walletIcon}
+                alt={accountSelectionTab.chooseWallet}
+                ml={3}
+              />
+            }
+          />
+          {evmAccountDropdownListGroup.map(group => {
+            const selectedItem = selectedEvmAccountsGroup.find(
+              a => a.assetId === group.assetId,
+            )?.accounts[0]?.__id;
+            return (
+              <Dropdown
+                key={group.assetId}
+                items={group.accounts}
+                selectedItem={selectedItem}
+                disabled={!selectedWallet}
+                searchText={parseLangTemplate(
+                  accountSelectionTab.chooseAccount,
+                  { assetName: coinList[group.assetId].name },
+                )}
+                placeholderText={parseLangTemplate(
+                  accountSelectionTab.chooseAccount,
+                  { assetName: coinList[group.assetId].name },
+                )}
+                onChange={id => {
+                  // use `handleSelectAccount` & `handleDisselectAccount` for multiselect
+                  const selectedAccount = evmAccountsGroup
+                    .find(a => a.assetId === group.assetId)
+                    ?.accounts.find(a => a.__id === id);
+                  if (selectedAccount) {
+                    setSelectedEvmAccounts([selectedAccount]);
+                  } else {
+                    setSelectedEvmAccounts([]);
+                  }
```

Comment:
- Don't export `set` functions from hooks.
- The only case where we should export set functions is if it's being used directly without and additional code.
- Define the `onChange` function in useEvmAccountDropdown

Reply:
- created `onChange` function in `useEvmAccountDropdown`.
- removed `setSelectedWallet` export from `useEvmAccountDropdown`.